### ISSUE TITLE
Stop burying activation errors in `CollectionModelBinder`

### ISSUE THREAD

**Issue by dougbu** [aspnet/Mvc#4193]
Inspired by [this discussion](https://github.com/aspnet/Mvc/pull/4172#discussion-diff-54319958), we had an offline conversation and decided to resolve activation error handling differences between `MutableObjectModelBinder` and `CollectionModelBinder` in favour of `MutableObjectModelBinder`. Problems activating the site's types should be handled the same whether those types are POCOs or collections. Since these problems are unrelated to information submitted in the request, a `ModelStateError` isn't appropriate.

Bottom line, the current silent failures in `CollectionModelBinder` just make the site developer's job more difficult. We'll let these `Exception`s propagate.

**Comment by dougbu** [comment]
d3c2463
